clang/lib/CodeGen/CGHLSLRuntime.h

```diff
@@ -165,10 +164,15 @@ class CGHLSLRuntime {
                                    BufferResBinding &Binding);
   void addConstant(VarDecl *D, Buffer &CB);
   void addBufferDecls(const DeclContext *DC, Buffer &CB);
+
+  llvm::Function *getOrCreateResourceBindingInitFn();
+  void generateInitResBindingsFuncBody();
+  void removeInitResBindingsFunc();
+
   llvm::Triple::ArchType getArch();
   llvm::SmallVector<Buffer> Buffers;
-
-  llvm::SmallVector<std::pair<const VarDecl *, llvm::GlobalVariable *>>
+  llvm::Function *InitResBindingsFunc;
+  llvm::SmallVector<std::pair<const NamedDecl *, llvm::GlobalVariable *>>
```

Comment:
Changing this from `VarDecl*` to `NamedDecl*` as a preparation for including `HLSLBufferDecl` in this list as well (llvm/llvm-project#113514).